Tests/XcbeautifyLibTests/ParsingTests.swift

```diff
@@ -175,4 +175,54 @@ final class ParsingTests: XCTestCase {
         // There's a regression whenever `uncapturedOutput` is greater than the current magic number.
         XCTAssertEqual(uncapturedOutput, 2)
     }
+
+    func testSwiftCompile() throws {
+        let url = Bundle.module.url(forResource: "swiftcompile-failure-with-stack", withExtension: "txt")!
+        let logContent = try String(contentsOf: url)
+        var buildLog = logContent.components(separatedBy: .newlines)
+        let parser = Parser()
+
+        var parsedLines: [any CaptureGroup] = []
+
+        while !buildLog.isEmpty {
+            let line = buildLog.removeFirst()
+            if let parsedLine = parser.parse(line: line) {
+                parsedLines.append(parsedLine)
+            }
+        }
+
+        func parseLine<T: CaptureGroup>(
+            _ line: any CaptureGroup,
+            to type: T.Type,
+            check: (T) -> Void
+        ) {
+            if let parsedLine = line as? T {
+                check(parsedLine)
+            } else {
+                XCTFail("Expected \(type), but got \(Swift.type(of: line))")
+            }
+        }
+
+        parseLine(parsedLines.removeFirst(), to: SwiftCompileFailedCaptureGroup.self) {
+            XCTAssertEqual(
+                $0.wholeError,
+                "Command SwiftCompile failed with a nonzero exit code"
+            )
+        }
+        parseLine(parsedLines.removeFirst(), to: SwiftCompileStackDumpHeaderCaptureGroup.self) {
+            XCTAssertEqual($0.wholeLine, "Stack dump:")
+        }
+
+        while let line = parsedLines.first {
+            parseLine(line, to: SwiftCompileStackDumpCaptureGroup.self) {
+                XCTAssertTrue(
+                    $0.wholeLine.contains {
+                        CharacterSet.decimalDigits.contains($0.unicodeScalars.first!)
+                    }
+                )
+            }
+            parsedLines.removeFirst()
+        }
+        XCTAssertTrue(parsedLines.isEmpty)
+    }
```

Comment:
@cpisciotta I think some behaviour tests would be better.

Like whole xcbeatify+terminal renderer with input and output. Snapshot-like tests. Because even if these two tests pass - I still won't see output until I support capture groups in formatter and outputrendering.